### PR TITLE
feat: improve `fromEvmAddress` developer experience

### LINKED ISSUE #1262: Implement utility method for zero-padding EVM addresses in the Address class
When converting an EVM address to a Fuel address, it's essential to pad zeros.

To simplify this process we need to add utility methods within the `Address` class that will automatically handle this zero-padding for us. This enhancement will reduce potential errors for all applications interacting with the EVM address.

### PR DESCRIPTION
**Pull request by danielbate**
Our current implementation of the `fromEvmAddress` method expects a wrapped and padded Evm Address to return an `Address`. As the EVM returns address unpadded, users would have to pad then wrap it to be compatible with the Sway/TS type.

To improve this experience, the `fromEvmAddress` method now accepts and unwrapped/unpadded string and will validate it to instantiate an `Address`. The `toEvmAddress` method has been renamed `toWrappedEvmAddress` to adhere to the above change.

Closes #1262

### CODE REVIEW

File: packages/address/src/address.ts [hunk 1]
```diff
@@ -98,7 +100,7 @@ export default class Address extends AbstractAddress {
    *
    * @returns The `bech32Address` property as an {@link EvmAddress | `EvmAddress`}
    */
-  toEvmAddress(): EvmAddress {
+  toWrappedEvmAddress(): EvmAddress {
```
Comment (Braqzen) [inline packages/address/src/address.ts:103]: I'm unsure about the name change. Is there a reason to change it?

File: packages/address/src/utils.ts [hunk 2]
```diff
@@ -73,6 +73,15 @@ export function isPublicKey(address: string): boolean {
   return (address.length === 130 || address.length === 128) && /(0x)?[0-9a-f]{128}$/i.test(address);
 }
 
+/**
+ * Determines if a given string is in EVM Address format
+ *
+ * @hidden
+ */
+export function isEvmAddress(address: string): boolean {
+  return (address.length === 42 || address.length === 40) && /(0x)?[0-9a-f]{40}$/i.test(address);
```
[the remark this answers is not in the document]
Comment (danielbate) [reply packages/address/src/utils.ts:82]: We do the same to the b256 address format so I gave the same leniency here. If we are saying that an Evm Address will always be `0x `prefixed then I can remove that leniency here.

File: packages/address/src/address.ts (hunk 1 above)
Comment (danielbate) [reply packages/address/src/address.ts:103]: I was trying to create some distinction between an Evm Address, being the unwrapped 20 byte address that `fromEvmAddress` takes as a function parameter, and the wrapped 256 bit Evm Address Sway type. 🤔

Comment (danielbate) [reply packages/address/src/address.ts:103]: That's a good point, I'll remove the infix

**Review by Dhaiwat10 — APPROVED** [review]
🪄

**Review by Braqzen — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

File: packages/address/src/utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (danielbate) [reply packages/address/src/utils.ts:82]: This is already captured in #1148

File: packages/address/src/address.test.ts [hunk 3]
```diff
@@ -128,6 +130,36 @@ describe('Address utils', () => {
...
+    expect(result).toBeFalsy();
+  });
+
+  test('isEvmAddress (too short)', () => {
+    const result = utils.isEvmAddress('0x123');
+
+    expect(result).toBeFalsy();
+  });
+
+  test('isEvmAddress (no hex prefix)', () => {
```
[the remark this answers is not in the document]
Comment (danielbate) [reply packages/address/src/address.test.ts:157]: This is captured in #1148

**Pull request merged**